testament/tester.nim

```diff
@@ -565,10 +565,12 @@ proc main() =
     p.next
     processCategory(r, cat, p.cmdLineRest.string, testsDir, runJoinableTests = false)
   of "r", "run":
-    let (dir, file) = splitPath(p.key.string)
-    let (_, subdir) = splitPath(dir)
-    var cat = Category(subdir)
-    processSingleTest(r, cat, p.cmdLineRest.string, file)
+    # at least one directory is required in the path, to use as a category name
+    let pathParts = split(p.key.string, {DirSep, AltSep})
```

Comment:
can a proc in os.nim be used ?

Reply:
None that I can find. The closest is splitPath(), but I need a short head, not a short tail.

Reply:
ya closest would be `iterator parentDirs(path: string; fromRoot = false; inclusive = true): string {...}` but in your case it may be inconvenient... maybe a rootDir would be needed